### PR TITLE
Support latest patch levels for HDP and CDH

### PR DESCRIPTION
**Pull request by wolf31o2**
This ensures support for the latest versions of HDP and CDH and sets the defaults to the latest patch level for the distributions. Also, switches big `if...elsif...elsif...end` into a case.

- HDP
  - 2.2.4.4 and 2.2.6.0 for HDP 2.2
  - 2.1.15.0 for HDP 2.1 (default)
- CDH
  - 5.3.4 (default)

### CODE REVIEW

File: recipes/repo.rb [hunk 1]
```diff
@@ -31,9 +31,9 @@
 # Set defaults for version, based on distribution
 node.default['hadoop']['distribution_version'] =
   if node['hadoop']['distribution'] == 'hdp'
-    '2.1.7.0'
+    '2.1.15.0'
   elsif node['hadoop']['distribution'] == 'cdh'
-    '5.3.2'
+    '5.3.4'
```
Comment (dereklwood) [inline recipes/repo.rb]: this should be 5.3.5

**Comment by dereklwood** [comment]
one cdh release behind, otherwise :+1:

File: recipes/repo.rb [hunk 2]
```diff
@@ -52,18 +52,20 @@
   when '2.0.4.0', '2.1.1.0', '2.2.0.0'
     hdp_version = node['hadoop']['distribution_version']
     hdp_update_version = nil
-  when '2.1.2.0', '2.1.2.1', '2.1.3.0', '2.1.4.0', '2.1.5.0', '2.1.7.0', '2.1.10.0'
+  when '2.1.2.0', '2.1.2.1', '2.1.3.0', '2.1.4.0', '2.1.5.0', '2.1.7.0', '2.1.10.0', '2.1.15.0'
```
Comment (duddun) [inline recipes/repo.rb]: dumb question: we know that there were no versions between 2.1.10.0 and 2.1.15.0?

Comment (wolf31o2) [inline recipes/repo.rb]: Yup. Dumb question... :smile: 

http://s3.amazonaws.com/public-repo-1.hortonworks.com/index.html#/HDP/centos6/2.x/updates

**Pull request merged**